samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -0,0 +1,389 @@
+            }
+        }
+    }
+
+    /// # Summary
+    /// OR oracle for an arbitrary number of qubits in query register
+    operation ApplyOrOracle (queryRegister : Qubit[], target : Qubit) : Unit is Adj {        
+        // x₀ ∨ x₁ = ¬ (¬x₀ ∧ ¬x₁)
+        // First, flip target if both qubits are in |0⟩ state
+        (ControlledOnInt(0, X))(queryRegister, target);
+        // Then flip target again to get negation
+        X(target);
+    }
+    
+    /// # Summary
+    /// Using Grover's search to find vertex coloring.
+    ///
+    /// # Input
+    /// ## numVertices
+    /// The number of Vertices in the graph
+    /// ## bitsPerColor
+    /// The number of bits per color
+    /// ## maxIterations
+    /// An estimate of the maximum iterations needed
+    /// ## oracle
+    /// The Oracle used to find solution
+    /// 
+    /// # Output
+    /// Int Array giving the color of each vertex
+    operation FindColorsWithGrover (numVertices : Int, bitsPerColor : Int, maxIterations : Int, 
+        oracle : ((Qubit[], Qubit) => Unit is Adj)) : Int[] {
+        // This task is similar to task 2.2 from SolveSATWithGrover kata, 
+        // (see https://github.com/microsoft/QuantumKatas/tree/master/SolveSATWithGrover)
+        // but the percentage of correct solutions is potentially higher.
+        mutable coloring = new Int[numVertices];
+
+        // Note that coloring register has the number of qubits that is 
+        // twice the number of vertices (bitsPerColor qubits per vertex).
+        using ((register, output) = (Qubit[bitsPerColor * numVertices], Qubit())) {
+            mutable correct = false;
+            mutable iter = 1;
+            repeat {
+                Message($"Trying search with {iter} iterations");
+                GroversAlgorithmLoop(register, oracle, iter);
+                let res = MultiM(register);
+                // to check whether the result is correct, apply the oracle to the 
+                // register plus ancilla after measurement
+                oracle(register, output);
+                if (MResetZ(output) == One) {
+                    set correct = true;
+                    // Read off coloring
+                    set coloring = MeasureColoring(bitsPerColor, register);
+                }
+                ResetAll(register);
+            } until (correct or iter > maxIterations)  
+            fixup {
+                set iter += 1;
+            }
+            if (not correct) {
+                fail "Failed to find a coloring";
+            }
+        }
+        return coloring;
+    }
+
+    /// # Summary
+    /// Grover loop implementation taken from SolveSATWithGrover kata.
+    /// (see https://github.com/microsoft/QuantumKatas/tree/master/SolveSATWithGrover)
+    ///
+    /// # Input
+    /// ## markingOracle
+    /// The oracle which will mark the valid solutions
+    operation OracleConverterImpl (markingOracle : ((Qubit[], Qubit) => Unit is Adj), 
```

Comment:
```suggestion
    operation ApplyPhaseOracle (oracle : ((Qubit[], Qubit) => Unit is Adj), 
```

Reply:
Just a suggestion for renaming - this operation applies the oracle in the X basis and then converts the target back to the Z basis, so you're effectively applying the oracle to the phase of the qubit, so I thought "ApplyPhaseOracle" would make sense. Feel free to rename if you prefer something else.